helix-term/src/ui/editor.rs

```diff
@@ -288,16 +288,25 @@ impl EditorView {
             .find_scope_index("ui.cursor")
             .unwrap_or(selection_scope);
 
-        let cursor_scope = match mode {
+        let mode_cursor_scope = match mode {
             Mode::Insert => theme.find_scope_index("ui.cursor.insert"),
             Mode::Select => theme.find_scope_index("ui.cursor.select"),
-            Mode::Normal => Some(base_cursor_scope),
-        }
-        .unwrap_or(base_cursor_scope);
+            Mode::Normal => None,
+        };
+
+        let cursor_scope = if theme.find_scope_index("ui.cursor.primary").is_some() {
+            base_cursor_scope
+        } else {
+            mode_cursor_scope.unwrap_or(base_cursor_scope)
+        };
+
+        let primary_cursor_scope =
+            if let Some(base_primary_cursor_scope) = theme.find_scope_index("ui.cursor.primary") {
+                mode_cursor_scope.unwrap_or(base_primary_cursor_scope)
+            } else {
+                cursor_scope
+            };
```

Comment:
I'm a bit confused by the conditionals here.

Reply:
this chooses between `mode_cursor_scope` and `base_cursor_scope`/`base_primary_cursor_scope` depending on whether `ui.cursor.primary` is present.

I'm considering making a new PR that cleans up the code to make all of the defaults and fallbacks clearer, and adding code for `ui.cursor.{insert,select}.primary` over it.